cmd/ecs-discovery/main.go

```diff
@@ -164,7 +164,8 @@ func (t *AugmentedTask) ExporterInformation() []*PrometheusTaskInfo {
 			return ret
 		}
 		for _, iface := range t.EC2Instance.NetworkInterfaces {
-			if iface.PrivateIpAddress != nil && *iface.PrivateIpAddress != "" {
+			if iface.PrivateIpAddress != nil && *iface.PrivateIpAddress != "" &&
+				strings.HasPrefix(*iface.Groups[0].GroupName, "mon-agg-ecs-") {
```

Comment:
An alternative way would be to check the interface type - it looks like it would be `interface` for the main interface, and `trunk` for the one that gets added by ECS - so we could do:

```golang
[...] && *iface.InterfaceType == "interface" 
```

Happy to do it this way instead if it feels better?

Reply:
> Happy to do it this way instead if it feels better?

![Same same really](https://media.giphy.com/media/VhRESt9d7S94rzOsaE/giphy.gif)

Down the line I'd like to explore if we can simplify this into a static file, I really don't think the system is dynamic enough to justify "service discovery" like this!